### PR TITLE
Question mark tooltips

### LINKED ISSUE #549: Deadline tooltips
**Describe the solution you'd like**
We have multiple deadline/time settings on the platform, not all of which are clear at first glance. Each time setting should be accompanied by an explanation of the exact impact. Be it in the form of a tooltip on hover or something similar. 

From the top of my head the following times need a definition:

- Course  (From, To)
- Assignment (From, Due, Lock)
- Deadline node (Date)

### PR DESCRIPTION
**Pull request by engelhamer**
## Description
Added question marks with tooltips to deadline dates and entry properties.

image

## Summary of changes
- Fixes #549

### CODE REVIEW

File: src/vue/src/components/format/FormatEditAssignmentDetailsCard.vue [hunk 1]
```diff
@@ -23,29 +23,65 @@
...
+                        :config="$root.flatPickrTimeConfig"/>
                 </b-col>
                 <b-col xl="4">
-                    <h2 class="field-heading">Due date</h2>
+                    <h2 class="field-heading">
+                        Due date
+                        <sup>
+                            <icon
+                                v-b-tooltip.hover
+                                title="Students are expected to have finished their assignment by this date"
```
Comment (mjvkeulen) [inline src/vue/src/components/format/FormatEditAssignmentDetailsCard.vue]: Same thing applies here, what are the actual implications?

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 2]
```diff
@@ -46,14 +46,32 @@
             </b-col>
         </b-row>
 
-        <h2 class="field-heading">Deadline</h2>
+        <h2 class="field-heading">
+            Deadline
+            <sup>
+                <icon
+                    v-b-tooltip.hover
+                    title="Students should have achieved the goal of this node by this date"
```
Comment (mjvkeulen) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: What are the implications if this does not happen? Is the student barred from making any further entries?

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 3]
```diff
@@ -66,7 +84,16 @@
             </div>
         </div>
         <div v-else-if="currentPreset.type === 'p'">
-            <h2 class="field-heading">Point Target</h2>
+            <h2 class="field-heading">
+                Point Target
+                <sup>
+                    <icon
+                        v-b-tooltip.hover
+                        title="The amount of points students should have achieved by the deadline of this node"
```
Comment (mjvkeulen) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: Ditto

File: src/vue/src/components/assignment/CreateAssignment.vue [hunk 4]
```diff
@@ -25,29 +25,65 @@
...
+                        :config="$root.flatPickrTimeConfig"/>
                 </b-col>
                 <b-col xl="4">
-                    <h2 class="field-heading">Due date</h2>
+                    <h2 class="field-heading">
+                        Due date
+                        <sup>
+                            <icon
+                                v-b-tooltip.hover
+                                title="Students are expected to have finished their assignment by this date"
```
[the remark this answers is not in the document]
Comment (engelhamer) [reply src/vue/src/components/assignment/CreateAssignment.vue]: Currently nothing at all..

[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assignment/CreateAssignment.vue]: No issues there right? Provided we communicate that these checks purely visual indicators

[the remark this answers is not in the document]
Comment (engelhamer) [reply src/vue/src/components/assignment/CreateAssignment.vue]: So you want to explicitly state that nothing happens? Wouldn't that be more confusing?

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue (hunk 2 above)
Comment (engelhamer) [reply src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: Again, nothing.

File: src/vue/src/components/assignment/CreateAssignment.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assignment/CreateAssignment.vue]: I think being more explicit should only help to clarify its impact. Confusion would only stem from how we word it right?

[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assignment/CreateAssignment.vue]: I realise that comment was a bit obvious... I feel stating that it only serves as a visual indicator of a passed deadline would convey its meaning more clearly.

File: src/vue/src/components/assignment/CreateAssignment.vue [hunk 5]
```diff
@@ -25,29 +25,65 @@
                 @content-update="form.assignmentDescription = $event"
                 :footer="false"
             />
-            <h2 class="field-heading">Points possible</h2>
+            <h2 class="field-heading">
+                Points possible
+                <sup>
+                    <icon
+                        v-b-tooltip.hover
+                        title="The maximum amount of points for this assignment"
```
Comment (mjvkeulen) [inline src/vue/src/components/assignment/CreateAssignment.vue]: Should we indicate entries can still be made even if the maximum number of points have been earned?

Comment (engelhamer) [reply src/vue/src/components/assignment/CreateAssignment.vue]: Sorry, changed the text on the other location but forgot to update it here.

Comment (engelhamer) [reply src/vue/src/components/assignment/CreateAssignment.vue]: It is `The amount of points that represents a perfect score for this assignment, excluding bonus points` there, do you think that covers the meaning?

**Pull request merged**